src/mame/atari/atari400.cpp

```diff
@@ -2232,180 +2289,145 @@ void a400_state::a400(machine_config &config)
 
 	A800_CART_SLOT(config, m_cartright, a800_right, nullptr);
+
+	// TODO: confirm all of the official config
+	// 8K release model, 48K possible here?
+	m_ram->set_default_size("16K");
+	m_ram->set_extra_options("8K,48K");
 }
 
 
 // memory map A800 + PAL screen + Right cartslot
-void a400_state::a800pal(machine_config &config)
+void a800_state::a800pal(machine_config &config)
 {
-	atari_common(config);
-
-	m_maincpu->set_addrmap(AS_PROGRAM, &a400_state::a400_mem);
-	TIMER(config, "scantimer").configure_scanline(FUNC(a400_state::a400_interrupt), "screen", 0, 1);
-
-	MCFG_MACHINE_START_OVERRIDE( a400_state, a800 )
+	a800(config);
 
 	config_pal_screen(config);
-//	m_screen->set_refresh_hz(antic_device::FRAME_RATE_50HZ);
-//	m_screen->set_size(antic_device::HWIDTH * 8, antic_device::TOTAL_LINES_50HZ);
-
-//	m_gtia->set_region(GTIA_PAL);
-
-	A800_CART_SLOT(config, m_cartright, a800_right, nullptr);
 }
 
-
-// memory map A600XL (same as 800XL but less RAM) + NTSC screen + MMU via PIA portB
-void a400_state::a600xl(machine_config &config)
+// memory map A1200XL+ MMU via PIA portB
+void a1200xl_state::atari_xl_common(machine_config &config)
 {
 	atari_common(config);
 
-	m_maincpu->set_addrmap(AS_PROGRAM, &a400_state::a600xl_mem);
-	TIMER(config, "scantimer").configure_scanline(FUNC(a400_state::a800xl_interrupt), "screen", 0, 1);
+	TIMER(config, "scantimer").configure_scanline(FUNC(a1200xl_state::xl_interrupt), "screen", 0, 1);
 
 	m_pokey->pot_r<4>().set_constant(0xff);
 	m_pokey->pot_r<5>().set_constant(0xff);
 	m_pokey->pot_r<6>().set_constant(0xff);
 	m_pokey->pot_r<7>().set_constant(0xff);
 
 	m_pia->readpb_handler().set_constant(0x83);
-	m_pia->writepb_handler().set(FUNC(a400_state::a600xl_pia_pb_w));
-
-	MCFG_MACHINE_START_OVERRIDE( a400_state, a800xl )
+	m_pia->writepb_handler().set(FUNC(a1200xl_state::pia_portb_w));
 
 	config_ntsc_screen(config);
 
-//	m_screen->set_refresh_hz(antic_device::FRAME_RATE_60HZ);
-//	m_screen->set_size(antic_device::HWIDTH * 8, antic_device::TOTAL_LINES_60HZ);
-
-//	m_gtia->set_region(GTIA_NTSC);
-
 	m_ram->set_default_size("16K");
 
 	config.device_remove("ctrl3");
 	config.device_remove("ctrl4");
 }
 
-// memory map A800XL + NTSC screen + MMU via PIA portB
-void a400_state::a800xl(machine_config &config)
+void a1200xl_state::a1200xl(machine_config &config)
 {
-	atari_common(config);
-
-	m_maincpu->set_addrmap(AS_PROGRAM, &a400_state::a800xl_mem);
-	TIMER(config, "scantimer").configure_scanline(FUNC(a400_state::a800xl_interrupt), "screen", 0, 1);
-
-	m_pokey->pot_r<4>().set_constant(0xff);
-	m_pokey->pot_r<5>().set_constant(0xff);
-	m_pokey->pot_r<6>().set_constant(0xff);
-	m_pokey->pot_r<7>().set_constant(0xff);
+	atari_xl_common(config);
 
-	m_pia->readpb_handler().set_constant(0x83);
-	m_pia->writepb_handler().set(FUNC(a400_state::a800xl_pia_pb_w));
+	m_maincpu->set_addrmap(AS_PROGRAM, &a1200xl_state::a1200xl_mem);
 
-	MCFG_MACHINE_START_OVERRIDE( a400_state, a800xl )
+	m_pokey->pot_r<4>().set_ioport("J1");
```

Comment:
Note: this will read 0x00 on initial boot no matter the settings in here, I wonder if we have to initialize Pokey pot values somehow (without a reset line?).
There's also this lovely portion in Pokey device_start:
https://github.com/mamedev/mame/blob/f5eeb2681d2cc1af975cd5cbd161b78644d68d51/src/devices/sound/pokey.cpp#L260